src/sdk/autodetectcompilers.cpp

```diff
@@ -12,137 +12,325 @@
+                    }
+                }
+            }
+            else
+            {
+                compilerListItem.status = "No compiler";
+                compilerListItem.compilerPath = "N/A";
+                compilerListItem.detected = true;
+                compilerListItem.colorHighlight = CompilerHighlightColor::colorHighlightGreen;
             }
 
-            if (highlight == 1)
-                list->SetItemBackgroundColour(idx, *wxRED);
-            else if (highlight == -1)
-                list->SetItemTextColour(idx, *wxLIGHT_GREY);
+            if (defaultCompilerID.IsSameAs(currentCompilerID))
+            {
+                wxStaticText* controlLblDefCompiler = XRCCTRL(*this, "lblDefCompiler", wxStaticText);
+                if (compilerListItem.detected)
+                {
+                    wxColour  colorTextDefault =  XRCCTRL(*this, "ID_STATICTEXT1", wxStaticText)->GetForegroundColour();
+                    wxColour  colorBackgroundDefault =  XRCCTRL(*this, "ID_STATICTEXT1", wxStaticText)->GetBackgroundColour();
+
+                    controlLblDefCompiler->SetLabel(compilerListItem.compilerName);
+                    controlLblDefCompiler->SetForegroundColour(colorTextDefault);
+                    controlLblDefCompiler->SetBackgroundColour(colorBackgroundDefault);
+                }
+                else
+                {
+                    controlLblDefCompiler->SetLabel(wxString::Format(_("INVALID: %s"),compilerListItem.compilerName));
+                    controlLblDefCompiler->SetForegroundColour(wxColour(*wxWHITE));
+                    controlLblDefCompiler->SetBackgroundColour(wxColour(*wxRED));
+                }
+            }
+
+            m_CompilerList.push_back(compilerListItem);
         }
-        // Resize columns so one can read the whole stuff:
-        list->SetColumnWidth(0, wxLIST_AUTOSIZE);
-        list->SetColumnWidth(1, wxLIST_AUTOSIZE);
+        UpdateCompilerDisplayList();
     }
-
-    XRCCTRL(*this, "lblDefCompiler", wxStaticText)->SetLabel(CompilerFactory::GetDefaultCompiler()->GetName());
 }
 
 AutoDetectCompilers::~AutoDetectCompilers()
 {
     //dtor
 }
+
+bool AutoDetectCompilers::closeCheckOkay()
+{
+    wxString defaultCompiler = CompilerFactory::GetDefaultCompiler()->GetName();
+
+    for (std::vector<CompilerItem>::iterator it = m_CompilerList.begin(); it != m_CompilerList.end(); ++it)
+    {
+        // Find default compiler in the list and if it was not detected double check with the user that this is okay.
+        if (it->compilerName.IsSameAs(defaultCompiler) && !it->detected)
+        {
+            if (wxMessageBox("Are you sure you want to configure CodeBlocks for an invalid compiler?", "",wxYES_NO) == wxNO)
+            {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+void AutoDetectCompilers::OnClose(wxCloseEvent& event)
+{
+    if ( event.CanVeto())
+    {
+        if (!closeCheckOkay())

```

Comment:
I don't like this. If this review passes, I'll just remove this code.
Forcing people to select a default compiler is not a good idea.
Probably wouldn't use this compiler once.

Reply:
They do not anymore.

Reply:
You've committed the too many message boxes crime. Nobody reads those, especially people new to the software which just want to type some code.